src/Mvc/Mvc.Razor/src/RazorPageBase.cs

```diff
@@ -72,25 +74,30 @@ public virtual TextWriter Output
     public dynamic ViewBag => ViewContext?.ViewBag!;
 
     /// <inheritdoc />
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
     public bool IsLayoutBeingRendered { get; set; }
 
     /// <inheritdoc />
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
     public IHtmlContent? BodyContent { get; set; }
 
     /// <inheritdoc />
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
     public IDictionary<string, RenderAsyncDelegate> PreviousSectionWriters { get; set; } = default!;
 
     /// <summary>
     /// Gets or sets a <see cref="System.Diagnostics.DiagnosticSource"/> instance used to instrument the page execution.
     /// </summary>
     [RazorInject]
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
     public DiagnosticSource DiagnosticSource { get; set; } = default!;
 
     /// <summary>
     /// Gets the <see cref="System.Text.Encodings.Web.HtmlEncoder"/> to use when this <see cref="RazorPage"/>
     /// handles non-<see cref="IHtmlContent"/> C# expressions.
     /// </summary>
     [RazorInject]
+    [DebuggerBrowsable(DebuggerBrowsableState.Never)]
```

Comment:
@JamesNK does this also hide it from the raw view?

Reply:
There isn't a raw view. As far as I know, `[DebuggerBrowsable(DebuggerBrowsableState.Never)]` = always hidden.

The debug view vs raw view happens when a `DebuggerProxyType` is added. The reason I haven't used a proxy here is Razor pages and Controllers are inherited from and have custom properties. For example, in the screenshots for controller above, the `Id` property is user added. A proxy is good when there is a set of known fields, but in this case there can user-added properties on user classes, and we don't want to hide them.

Now, what we could do is change these auto properties into getters/setters backed by private fields. Someone can click on "Show private fields" in debug UI and see what `_isLayoutBeingRendered` is set to.

Reply:
I'm not sure why we would ever hide anything that we can't get back in the debugger. This makes lives hard for us as well.

Reply:
My suggestion makes the values available as a field:

> One simple solution could be to change these auto properties into getters/setters backed by private fields. Someone can click on "Show private fields" in debug UI and see what `_isLayoutBeingRendered` is set to.

We get the best of both worlds:
1. We avoid cluttering the debugger UI with things the user doesn't care about, such as ProblemDetailsFactory, ModelMetadataProvider, etc. I'm guessing these properties were made public so they could be injected when the page/controller was created. The downside is there are a lot of public properties on base classes that no one cares about. The [RazorPage debug UI screenshot above](https://user-images.githubusercontent.com/303201/252911080-674364b4-1f09-4da7-aee8-ed131feaa094.png) is a shitshow.
2. The values are still debuggable as private fields.

Granted, it's not as clean as making everything accessible as it was with "Raw view", but for the 99.9% use case of a customer debugging their app, it's a solid improvement.

Reply:
@jaredpar Do you have any suggestions here? Or can you recommend some other debugger experts to take a look?

Reply:
Don't have any other recommendations. Usually the solution for this would be a `DebuggerTypeProxy` but this is a base type so don't think that will work. 

> There isn't a raw view. As far as I know, [DebuggerBrowsable(DebuggerBrowsableState.Never)] = always hidden.

Matches my understanding: never means never. Checked and it's even the case if you force the debugger into full raw view mode.

Reply:
Lets avoid using never then.